cmd/kogito/command/flag/infraresource_flag.go

```diff
@@ -39,14 +39,13 @@ func AddInfraResourceFlags(command *cobra.Command, flags *InfraResourceFlags) {
 
 // CheckInfraResourceArgs validates the InfraResourceFlags flags
 func CheckInfraResourceArgs(flags *InfraResourceFlags) error {
-	if len(flags.Kind) == 0 {
-		return fmt.Errorf("kind can't be empty")
-	}
-	if len(flags.APIVersion) == 0 {
-		return fmt.Errorf("apiVersion can't be empty")
-	}
-	if len(flags.ResourceName) == 0 {
-		return fmt.Errorf("resource-name can't be empty")
+	if len(flags.Kind) > 0 {
+		if len(flags.APIVersion) == 0 {
+			return fmt.Errorf("apiVersion can't be empty")
+		}
+		if len(flags.ResourceName) == 0 {
+			return fmt.Errorf("resource-name can't be empty")
+		}
```

Comment:
you check first the resourcename in [here](https://github.com/kiegroup/kogito-operator/pull/967/files#diff-fec05a480e45a71908bc3622422e0983fdd2fdb7f97de5ad34f0542940f41b47R24) so I guess we should have more something like:
```suggestion
	if len(flags.ResourceName) > 0 {
		if len(flags.APIVersion) == 0 {
			return fmt.Errorf("apiVersion can't be empty")
		}
		if len(flags.Kind) == 0 {
			return fmt.Errorf("kind can't be empty")
		}
```
or ?

Reply:
done